### ISSUE TITLE
Rename master branch

### ISSUE THREAD

**Issue by billsacks** [ESCOMP/CTSM#1048]
The `master` term can be offensive, and so GitHub and others are encouraging switching to a more neutral term. `main` seems to be a good choice; some others would be `latest`, `trunk`, `stable` or `develop`. It sounds like GitHub will be publishing guidance on how to accomplish this soon. Otherwise, a quick search turned up these recent guides:

https://dev.to/rhymu8354/git-renaming-the-master-branch-137b

https://www.hanselman.com/blog/EasilyRenameYourGitDefaultBranchFromMasterToMain.aspx

I'm calling this a "documentation" issue because the pieces needed are probably more on the documentation side than anywhere else, and I don't think this will require any actual code changes.

**Comment by billsacks** [comment]
@ckoven thanks for suggesting this idea on the FATES side.

**Comment by wwieder** [comment]
Yep, let's make this move & make the transition as smooth as possible for cloned repos.

**Comment by billsacks** [comment]
https://github.com/github/renaming

**Comment by ekluzek** [comment]
As part of this we should also change our documentation (README files and the like) that refer to the master branch. Also likely in our wiki pages as well.

**Comment by rgknox** [comment]
I was reading the second guide you posted @billsacks , thanks.  
It seems like this will only require two commands to make the change.

**Comment by ekluzek** [comment]
I think github.com has done all of the changes they plan to implement to help this change. They make it easy on the web-interface, and it automatically forwards branch protections and even lets people know about the change on a "push". The web-interface also lets you know what you need to do. When you go to your fork it does for example point out that you need to change it. And the first time you go to a repository with the change there's a pop up that tells you about the change.

Clones and forks do have to change. The commands for us in git on a clone are like this (github also tells you about the need to do this)...

```
git branch -m master main
git fetch escomp
git branch -u escomp/main main
```
(Note the details of above does depend on how you have named your remotes, this assumes you use the escomp convention)

I've done this in PTCLM, just to see how easy it was and what had to be done.

**Comment by billsacks** [comment]
Awesome, thanks @ekluzek for looking into this more. I have edited the top-level comment to put a checklist of things I can think of that need to be done. Feel free to edit it and add anything else you can think of.

**Comment by wwieder** [comment]
Is this something we need to help communicate to model users / developers, or can new forks & PRs just merge these changes without too much disruption?

**Comment by billsacks** [comment]
> Is this something we need to help communicate to model users / developers, or can new forks & PRs just merge these changes without too much disruption?

I think it depends. My understanding is that the main disruption – and the need for the commands @ekluzek posted yesterday – comes when you have an existing local clone and want to do anything with the master/main branch in that clone. If, on the other hand, you either (a) have a local clone but aren't doing anything with the master/main branch (which means never updating your local branch to merge in the latest changes from master/main), or (b) are making a new clone for your work, then my understanding is that you can remain blissfully unaware of this change.

**Comment by rgknox** [comment]
Lets say a user wants to add a feature.  My first steps would be to update master/main so I have the most recent base to build my feature upon.  So when a user tries to pull in updates to master (on their local clone), github will let them know that master is deprecated and to change the name to main?  I suppose I could make the change on my fork and see what happens.

**Comment by ekluzek** [comment]
Yes, I think the problem is that you always want your branch to be updated to the latest on the main development branch. If you aren't updating then I suppose you could remain unaware. But, most likely you do want to stay up to date, so you will. I don't think you'll get the newest tags without correcting the name from master to main.

What I saw was that a push to master let me know about the change, but a pull to master failed. Also the web-interface tells me about the change and what I have to do for it. And it tells me both on the ESCOMP fork and my own fork.

So I think we will want to communicate to users a bit about this. It's not completely critical since github also informs you. But, CESM is likely to make this same change and they might communicate to CESM users, so we probably want to coordinate with those efforts as well.

**Comment by ekluzek** [comment]
The plan today is that we'll do this right after we make the ctsm5.1.0 tag with the start of the ctsm5.2.dev series. That should be fairly soon now.

**Comment by ekluzek** [comment]
This comes up again, as we want to move to a development branch/main branch kind of workflow.

**Comment by ekluzek** [comment]
Examples of current files that need to be updated here:
```
.github/PULL_REQUEST_TEMPLATE.md:
.github/workflows/docker-image-build-publish.yml
.github/workflows/docs-build-and-deploy.yml
CONTRIBUTING.md:
README
doc/.ChangeLog_template
doc/source/lilac/obtaining-building-and-running/obtaining-and-building-ctsm.rst
doc/source/users_guide/setting-up-and-running-a-case/customizing-the-clm-configuration.rst
doc/substitutions.py
```
